includes/rest-api/class-wp-rest-stories-controller.php

```diff
@@ -0,0 +1,131 @@
+	/**
+	 * Prepares a single story for create or update. Add post_content_filtered field to save/insert.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_story = parent::prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_story ) ) {
+			return $prepared_story;
+		}
+
+		if ( isset( $request['content_filtered'] ) ) {
+			$prepared_story->post_content_filtered = $request['content_filtered'];
+			if ( isset( $request['content_filtered']['raw'] ) ) {
+				$prepared_story->post_content_filtered = $request['content_filtered']['raw'];
+			}
+			$prepared_story->post_content_filtered = maybe_serialize( $prepared_story->post_content_filtered );
+		}
+
+
+		return $prepared_story;
+	}
+
+	/**
+	 * Prepares a single story output for response. Add post_content_filtered field to output.
+	 *
+	 *
+	 * @param WP_Post $post Post object.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ) {
+		$response = parent::prepare_item_for_response( $post, $request );
+		$fields   = $this->get_fields_for_response( $request );
+		$data     = $response->get_data();
+		$schema   = $this->get_item_schema();
+
+		if ( in_array( 'content_filtered', $fields, true ) ) {
+			$post_content_filtered    = maybe_unserialize( $post->post_content_filtered );
+			$data['content_filtered'] = array(
+				'raw' => rest_sanitize_value_from_schema( $post_content_filtered, $schema['properties']['content_filtered']['properties']['raw'] ),
+			);
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->filter_response_by_context( $data, $context );
+		$links   = $response->get_links();
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+		foreach ( $links as $rel => $rel_links ) {
+			foreach ( $rel_links as $link ) {
+				$response->add_link( $rel, $link['href'], $link['attributes'] );
+			}
+		}
+
+		/**
+		 * Filters the post data for a response.
+		 *
+		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
+		 *
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param WP_Post $post Post object.
+		 * @param WP_REST_Request $request Request object.
+		 */
+		return apply_filters( "rest_prepare_{$this->post_type}", $response, $post, $request );
+	}
+
+	/**
+	 * Retrieves the attachment's schema, conforming to JSON Schema.
+	 *
+	 *
+	 * @return array Item schema as an array.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+		$schema = parent::get_item_schema();
+
+		$schema['properties']['content_filtered'] = array(
```

Comment:
As opposed to `content_filtered` what about something that avoids exposing some of the internals, that we're using `post_content_filtered`? In other words, perhaps this should rather be something like `story_data`.

It would be kinda nice if the field were added alongside `content.raw` and `content.rendered`, but I understand there is a limitation in the REST API which doesn't allow for this. See https://core.trac.wordpress.org/ticket/47779

Reply:
I think that is confusing. If we wanted it named `story_data` we should have gone with register meta. The field should reflect what it maps to in core.

Reply:
Why? The REST API intentionally diverged from what core uses internally in `wp_insert_post()` because the legacy naming was bad. For example, the REST API uses `content` instead of `post_content` and `slug` instead of `post_name`. It seems totally fine to use a more specific field name than `content_filtered`. The REST API client shouldn't care at all where the data is being stored.

Reply:
It is confusing for new developer, as they would have to look through the code to see what field that maps too. I don't see the benefit of hiding the field name here.

Reply:
It's not hiding the field. It's using a more semantic field name for the REST API client. If a developer wants to find how the data is stored in PHP, they'd have to look at the controller anyway. When doing so, they'd see how it maps.

Reply:
Field name changed - https://github.com/ampproject/amp-wp/pull/3871/commits/25c14233ebe86581b6ed385871fe85ef7784f669